CI failure [msrv and beta]

* rust beta toolchain got upgraded to 1.81.x:

https://github.com/open-telemetry/opentelemetry-rust/actions/runs/10063464987/job/27818273607?pr=1955
```
   Compiling stress v0.1.0 (/home/runner/work/opentelemetry-rust/opentelemetry-rust/stress)
error: struct `SamplingStrategyParameters` is never constructed
   --> opentelemetry-sdk/src/trace/sampler/jaeger_remote/remote.rs:76:19
    |
76  | pub(crate) struct SamplingStrategyParameters {
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
note: the lint level is defined here
   --> opentelemetry-sdk/src/lib.rs:121:24
    |
121 | #![cfg_attr(test, deny(warnings))]
    |                        ^^^^^^^^
    = note: `#[deny(dead_code)]` implied by `#[deny(warnings)]`

error: could not compile `opentelemetry_sdk` (lib test) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
Error: Process completed with exit code 101.

```

and 

* msrv:

https://github.com/open-telemetry/opentelemetry-rust/actions/runs/10063464987/job/27818273359?pr=1955
```
  Downloaded linux-raw-sys v0.4.14
error: package `tokio-macros v2.4.0` cannot be built because it requires rustc 1.70 or newer, while the currently active rustc version is 1.65.0
Either upgrade to rustc 1.70 or newer, or use
cargo update -p tokio-macros@2.4.0 --precise ver
where `ver` is the latest version of `tokio-macros` supporting rustc 1.65.0
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -42,7 +42,7 @@ temp-env = "0.3.6"
 thiserror = { version = "1", default-features = false }
 tonic = { version = "0.12", default-features = false }
 tonic-build = "0.12"
-tokio = { version = "1.38.0", default-features = false } #1.39 needs msrv bump to rustc 1.70
+tokio = { version = "~1.38.0", default-features = false } #1.39 needs msrv bump to rustc 1.70
 tokio-stream = "0.1.1"
 tracing = { version = "0.1", default-features = false }
 tracing-core = { version = "0.1", default-features = false }

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -42,7 +42,7 @@ temp-env = "0.3.6"
 thiserror = { version = "1", default-features = false }
 tonic = { version = "0.12", default-features = false }
 tonic-build = "0.12"
-tokio = { version = "1", default-features = false }
+tokio = { version = "1.38.0", default-features = false } #1.39 needs msrv bump to rustc 1.70
 tokio-stream = "0.1.1"
 tracing = { version = "0.1", default-features = false }
 tracing-core = { version = "0.1", default-features = false }

--- a/opentelemetry-sdk/src/trace/sampler/jaeger_remote/mod.rs
+++ b/opentelemetry-sdk/src/trace/sampler/jaeger_remote/mod.rs
@@ -1,4 +1,5 @@
 mod rate_limit;
+#[allow(dead_code)]
 mod remote;
 mod sampler;
 mod sampling_strategy;

